src/views/undo.c

```diff
@@ -43,21 +45,30 @@ void dt_undo_cleanup(dt_undo_t *self)
   dt_pthread_mutex_destroy(&self->mutex);
 }
 
-void dt_undo_record(dt_undo_t *self, dt_view_t *view, dt_undo_type_t type, dt_undo_data_t *data,
-                    void (*undo)(dt_view_t *view, dt_undo_type_t type, dt_undo_data_t *item))
+static void _free_undo_data(void *p)
+{
+  dt_undo_item_t *item = (dt_undo_item_t *)p;
+  if (item->free_data) item->free_data(item->data);
+  g_free(item);
```

Comment:
Sure this should be `g_free()`?
If we manage it, `malloc()`/`free()` is much better.